pkg/plugins/policies/core/matchers/dataplane.go

```diff
@@ -125,14 +125,15 @@ func filterGatewaysByZone(gateways []*core_mesh.MeshGatewayResource, dpp *core_m
 		return gateways
 	}
 	var filtered []*core_mesh.MeshGatewayResource
-	dppZone := dpp.GetMeta().GetLabels()[mesh_proto.ZoneTag]
+	dppZone, dppZoneOk := dpp.GetMeta().GetLabels()[mesh_proto.ZoneTag]
 	for _, gateway := range gateways {
 		gwOrigin, ok := gateway.GetMeta().GetLabels()[mesh_proto.ResourceOriginLabel]
 		if !ok || gwOrigin == string(mesh_proto.GlobalResourceOrigin) {
 			filtered = append(filtered, gateway)
 			continue
 		}
-		if gwZone, ok := gateway.GetMeta().GetLabels()[mesh_proto.ZoneTag]; ok && gwZone == dppZone {
+		gwZone, gwZoneOk := gateway.GetMeta().GetLabels()[mesh_proto.ZoneTag]
```

Comment:
Should we have some kind of helper method on Meta or Labels? I find it hard to know what the semantics are of missing zone tag here. Like why is the check on gwZoneOk now inverted?

Reply:
We inverted it since resources created in 2.7 only have the `kuma.io/origin` label but don’t have the `kuma.io/zone` label. This causes the object to not be recognized as a zone object. I think we could introduce a helper like `IsLocallyOriginated` to address this.